bugwarrior/services/__init__.py

```diff
@@ -481,22 +481,21 @@ def __repr__(self):
         return '<%s>' % self.__unicode__()
 
 
-class ServiceClient(object):
-    """ Abstract class responsible for making requests to service API's. """
-    @staticmethod
-    def json_response(response):
-        # If we didn't get good results, just bail.
-        if response.status_code != 200:
-            raise IOError(
-                "Non-200 status code %r; %r; %r" % (
-                    response.status_code, response.url, response.text,
-                ))
-        if callable(response.json):
-            # Newer python-requests
-            return response.json()
-        else:
-            # Older python-requests
-            return response.json
+def json_response(response):
+    """ Process a request's response to keep HTTP error handling consistent
+    accross services. """
+    # If we didn't get good results, just bail.
+    if response.status_code != 200:
+        raise IOError(
+            "Non-200 status code %r; %r; %r" % (
+                response.status_code, response.url, response.text,
+            ))
+    if callable(response.json):
+        # Newer python-requests
+        return response.json()
+    else:
+        # Older python-requests
+        return response.json
```

Comment:
This is all rather hypothetical. And I think that a common pattern doesn't necessarily imply shared code and shared code doesn't necessarily imply a common parent class.

But I can live with it either way, so if you feel strongly about it tell me and I'll revert the commit.

> you can look at their base class, read the docstring, and understand how their purpose is distinct from the other classes.

In an ideal world, I could just look at the subclass' docstring to understand its purpose, but I understand we don't live in such a world :wink:.